### ISSUE TITLE
Retry more often in RetryingProvider, with the exception of eth_estimateGas

### ISSUE THREAD

**Issue by tkporter** [abacus-network/abacus-monorepo#706]
Copying some messages I put in Discord:

We definitely see intermittent provider issues and they are causing agents to crash. Learnings:
1. It seems that providers using a reverse proxy like cloudflare can result in instances where cloudflare responds to us with (I assume?) a successful 2XX HTTP code, but the contents of the response include an error that cloudflare has gotten from the actual provider. E.g.:
```
<head><title>524 Origin Time-out</title></head>
<body bgcolor="white">
<center><h1>524 Origin Time-out</h1></center>
<hr><center>cloudflare-nginx</center>
</body>
</html>
```
I think we originally expected non-successful-HTTP codes to hit the `HttpClientError::ReqwestError(e)` branch here https://github.com/abacus-network/abacus-monorepo/blob/main/rust/chains/abacus-ethereum/src/retrying.rs#L111 but clearly that's not the case as we're instead hitting the `HttpClientError::SerdeJson { err, text }` branch

2. It's also possible for internal server errors to occur that are exposed not via HTTP status codes or by cloudflare, but as a JSON RPC response, e.g:
```
{"timestamp":"2022-07-07T08:50:32.857938Z","level":"WARN","fields":{"message":"JsonRpcError in retrying provider; not retrying.","error":"(code: -32603, message: Internal server error, data: None)"},"target":"abacus_ethereum::retrying","span":{"name":"get_finalized_block_number"},"spans":[{"name":"get_finalized_block_number"}]}
```

I think we should retry for HttpClientError::SerdeJson errors. I don't see the downside to retrying here.

As for provider-level errors, imo we should just retry `eth_call`s. These are responsible for the validator crashes we're seeing, and imo as we already have teams other than ourselves operating validators, we really don't want to have them be crashing (bc of the liveness risk that we don't have control over, but also because it just reflects poorly on us)

The big problem with retrying RPC-level errors is that we don't want to retry when estimating gas and the gas estimation fails. `eth_estimateGas` calls are done when attempting to process a message, and if the call being estimated reverts, the `eth_estimateGas` RPC will return an error and the `HttpClientError::JsonRpcError(e)` branch in our RetryingProvider will be matched. If we spend a bunch of time retrying gas estimation for a message that is just gonna revert each time, we're wasting time and potentially clogging the work queue.

Imo the easiest option is to retry for `HttpClientError::JsonRpcError(e)`s whenever the method is anything other than `eth_estimateGas`. If an `eth_estimateGas` call ever does error, either because the call reverts or because of some "Internal server error"-style error in the RPC response, we handle this gracefully anyways in the relayer processing logic.

TLDR:
* We should change RetryingProvider to retry in all conditions other than when there is an `HttpClientError::JsonRpcError(e)` for the `eth_estimateGas` method.

**Comment by tkporter** [comment]
There may be a good argument for treating `eth_sendTransaction` or `eth_sendRawTransaction` in the same way as I'm advocating for `eth_estimateGas`, curious to hear thoughts here
